test/listen.test.js

```diff
@@ -372,12 +371,19 @@ test('addresses getter', async t => {
   const localAddresses = await dns.lookup('localhost', { all: true })
   for (const address of localAddresses) {
     address.port = port
-    if (semver.lt(process.versions.node, '18.0.0')) {
+    if (typeof address.family === 'number') {
```

Comment:
Will core coerce `'4'` to `4` when setting `address.family`?

Reply:
`dns.lookup` should always return `number`.
only the `server.address()` will change from `4 | 6` back to string `IPv4 | IPv6`.